persistence/src/commonTest/kotlin/com/wire/kalium/persistence/backup/RestoreBackupTest.kt

```diff
@@ -0,0 +1,487 @@
+        assertEquals(userToBackup, restoredUsers)
+    }
+
+    @Test
+    fun givenBackupHasNoUsersAndUserDoes_whenRestoringBackup_thenThoseUsersAreRestored() = runTest {
+        // given
+        val usersPresent = userDatabaseDataGenerator.generateAndInsertUsers(10)
+
+        // when
+        userDatabaseBuilder.backupImporter.importFromFile(databasePath(backupUserIdEntity))
+
+        // then
+        val restoredUsers = userDatabaseBuilder.userDAO.getAllUsers().first()
+        assertEquals(usersPresent, restoredUsers)
+    }
+
+    @Test
+    fun givenBackupHasUniqueUsersAndUserDoes_whenRestoringBackup_thenThoseUsersAreRestored() = runTest {
+        // given
+        val usersPresent = userDatabaseDataGenerator.generateAndInsertUsers(100)
+        val userToBackup = backupDatabaseDataGenerator.generateAndInsertUsers(50)
+
+        // when
+        userDatabaseBuilder.backupImporter.importFromFile(databasePath(backupUserIdEntity))
+
+        // then
+        val restoredUsers = userDatabaseBuilder.userDAO.getAllUsers().first()
+        assertTrue(restoredUsers.containsAll(userToBackup))
+        assertEquals(usersPresent.size + userToBackup.size, restoredUsers.size)
+    }
+
+    @Test
+    fun givenBackupHasOverlappingUsersWithUserAlongWithUniqueOnes_whenRestoringBackup_thenOnlyUniqueOnesAReRestored() = runTest {
+        // given
+        val uniqueUsersAmount = 10
+        val uniqueBackupUsersAmount = 10
+
+        val uniqueUsers = userDatabaseDataGenerator.generateAndInsertUsers(uniqueUsersAmount)
+        val uniqueBackupUsers = backupDatabaseDataGenerator.generateAndInsertUsers(uniqueBackupUsersAmount)
+
+        uniqueBackupUsers.forEach { userEntity ->
+            backupDatabaseBuilder.userDAO.insertUser(userEntity)
+        }
+
+        // when
+        userDatabaseBuilder.backupImporter.importFromFile(databasePath(backupUserIdEntity))
+
+        // then
+        val restoredUsers = userDatabaseBuilder.userDAO.getAllUsers().first()
+        assertEquals(restoredUsers,  uniqueUsers + uniqueBackupUsers)
+        assertEquals(uniqueUsersAmount + uniqueBackupUsersAmount, restoredUsers.size)
+    }
+
+    @Test
+    fun givenBackupHasAssetsButUserNot_whenRestoringBackup_thenBackupAssetsAreRestored() = runTest {
+        // given
+        val assetsToBackup = backupDatabaseDataGenerator.generateAndInsertAssets(10)
+
+        // when
+        userDatabaseBuilder.backupImporter.importFromFile(databasePath(backupUserIdEntity))
+
+        // then
+        assetsToBackup.forEach { restoredAssetEntity ->
+            val asset = userDatabaseBuilder.assetDAO.getAssetByKey(restoredAssetEntity.key).first()
+            assertTrue(asset != null)
+            assertEquals(asset, restoredAssetEntity)
+        }
+    }
+
+    @Test
+    fun givenUserHasAssetButBackupNot_whenRestoringBackup_thenUserAssetAreRestored() = runTest {
+        // given
+        val assetsToRestore = userDatabaseDataGenerator.generateAndInsertAssets(10)
```

Comment:
I don't really understand why these are called `assetsToRestore`. These are the current assets that live inside the current user database right? so they are not really restored and nothing is really being done when importing.

Reply:
I guess it is a matter of the naming or how we see those assets, I see both backup data and user data as a data that needs to be restored, any suggestion how to rename it ?

Reply:
`currentUserAssets`?